metricflow/model/validations/validator_helpers.py

```diff
@@ -160,19 +180,92 @@ def as_readable_str(self, with_level: bool = True) -> str:
         )
 
 
-class ValidationError(ValidationIssue):
+class ValidationError(ValidationIssue, BaseModel):
     """An error that was found while validating the model."""
 
-    level = ValidationIssueLevel.ERROR
+    @property
+    def level(self) -> ValidationIssueLevel:  # noqa: D
+        return ValidationIssueLevel.ERROR
 
 
-class ValidationFatal(ValidationIssue):
+class ValidationFatal(ValidationIssue, BaseModel):
     """A fatal issue that was found while validation the model."""
 
-    level = ValidationIssueLevel.FATAL
+    @property
+    def level(self) -> ValidationIssueLevel:  # noqa: D
+        return ValidationIssueLevel.FATAL
 
 
-ValidationIssueType = Union[ValidationIssue, ValidationWarning, ValidationFutureError, ValidationError, ValidationFatal]
+ValidationIssueType = Union[ValidationWarning, ValidationFutureError, ValidationError, ValidationFatal]
+
+
+class ModelValidationResults(FrozenBaseModel):
+    """Class for organizating the results of running validations"""
+
+    warnings: Tuple[ValidationWarning, ...] = tuple()
+    future_errors: Tuple[ValidationFutureError, ...] = tuple()
+    errors: Tuple[ValidationError, ...] = tuple()
+    fatals: Tuple[ValidationFatal, ...] = tuple()
+
+    @property
+    def has_blocking_issues(self) -> bool:
+        """Does the ModelValidationResults have ERROR or FATAL issues"""
+        return len(self.errors) != 0 or len(self.fatals) != 0
+
+    @classmethod
+    def from_issues_sequence(cls, issues: Sequence[ValidationIssueType]) -> ModelValidationResults:
+        """Constructs a ModelValidationResults class from a list of ValidationIssues"""
+
+        warnings: List[ValidationWarning] = []
+        future_errors: List[ValidationFutureError] = []
+        errors: List[ValidationError] = []
+        fatals: List[ValidationFatal] = []
+
+        for issue in issues:
+            if issue.level == ValidationIssueLevel.WARNING:
+                warnings.append(issue)
```

Comment:
I'm surprised that there's no type issue here. `issue` is of type `ValidationIssueType` but warnings is a `List[ValidationWarning]`.

Reply:
I think it's because issue `ValidationIssueType` is a `Union[ValidationWarning, ValidationFutureError, ValidationError, ValidationFatal` and the `level` of each of those is hardcoded. I.e. a `ValidationWarning` will always and only have a `level` value of `ValidationIssueLevel.WARNING`. Thus mypy can do it's correct magic.